resolver/serviceResolver.go

```diff
@@ -413,11 +413,11 @@ func (resolver *serviceResolver) validateTxRequestReturningGuardian(
 		return core.GuardianInfo{}, nil, err
 	}
 
-	return resolver.verifyCodesReturningGuardian(userAddress, txs[0].GuardianAddr, userIp, code, secondCode)
+	return resolver.verifyCodesReturningGuardian(userAddress.AddressAsBech32String(), txs[0].GuardianAddr, userIp, code, secondCode)
 }
 
 func (resolver *serviceResolver) verifyCodesReturningGuardian(
-	userAddress sdkCore.AddressHandler,
+	userAddress string,
```

Comment:
not needed, if the above suggestion would be applied

Reply:
reverted